compaction/task_manager_module.cc

```diff
@@ -423,6 +423,10 @@ future<> table_major_keyspace_compaction_task_impl::run() {
 
 future<> cleanup_keyspace_compaction_task_impl::run() {
     co_await _db.invoke_on_all([&] (replica::database& db) -> future<> {
+        if (_flush_mode == flush_mode::compacted_tables) {
```

Comment:
This doesn't make sense.  The condition should be `if (_flush_mode == flush_mode::all_tables)`.
The `compacted_tables` flush mode means that we want to flush only those that are subject to the current (cleanup) compaction.

Reply:
And in the above case, when all tables are flushed here, we need to pass down to `shard_cleanup_keyspace_compaction_task_impl` `flush_mode::skip`, since no further flush is required downstream.

Reply:
probably i misinterpret the name of `cleanup_keyspace_compaction_task_impl`. does it imply that this is a compaction task? if it does, then `all_tables` should applies to the user case where we do flush **before** performing this task, no? see the other reply.

Reply:
no.  it is a flag instructing the called task what to do, not telling it what the caller has already done.